cmd/models/validate.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright © 2023 OpenFGA
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package models
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/spf13/cobra"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type validationResult struct {
+	ID        string     `json:"id,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	IsValid   bool       `json:"is_valid"`
+	Error     *string    `json:"error,omitempty"`
+}
+
+func validate(inputModel string) (string, error) {
+	model := &pb.AuthorizationModel{}
+
+	err := protojson.Unmarshal([]byte(inputModel), model)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse input due to %w", err)
```

Comment:
Instead of treating invalid JSON as special error, should we treat it the same as other invalid model?
i.e.,
```
{"is_valid":false,"error":"invalid json"}
```

Reply:
That makes sense, but it makes it inconsistent w/ the API.

The API returns different errors on invalid json vs invalid models

Reply:
One is invalid input, the other is syntactically valid input but that is an invalid model

Reply:
Will change that to your suggestion 👍🏽

Reply:
Changed